web/src/store/slice/requestedItemsSlice.ts

```diff
@@ -0,0 +1,40 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit"
+
+interface IRequestItem {
+  id: number,
+  itemId: number,
+  items: string,
+  itemCode: string
+  quantity: number,
+  units: string,
+  unitsId: number,
+  brand?: string,
+  specification: string,
+}
+
+const initialState: IRequestItem[] = []
```

Comment:
interface InitialState<T> {
    data: T 
    isFetching?: boolean
}
```suggestion
const initialState: InitialState<IRequestItem[]> = {
  data: [],
  isFetching: false
} 
```